### PR TITLE
component delivery

### PR DESCRIPTION
**Pull request by BinaryHB0916**

### CODE REVIEW

**Comment by BinaryHB0916** [comment]
PTAL @FogDong @hongchaodeng

File: docs/end-user/component-delivery.md [hunk 1]
```diff
@@ -2,4 +2,210 @@
...
+app-delivering-chart	redis-comp	helm      	      	running	healthy	      	2021-08-28 18:48:21 +0800 CST
+```
+
+We also see that the PHASE of the app-delivering-chart APP is running and the STATUS is healthy.
+
+For usage examples of Git repositories and OSS buckets, and detailed configuration information about them, please go to [Built-in Components][3] in the Admin's Guide.
+
+## Using Kustomize Componet
+
+KubeVela's `kustomize` component meets the needs of users to directly connect Yaml files and folders as component products. No matter whether your Yaml file/folder is stored in a Git warehouse or an OSS bucket, KubeVela can read and deliver it.
```
Comment (FogDong) [inline docs/end-user/component-delivery.md]:
```suggestion
KubeVela's `kustomize` component meets the needs of users to directly connect YAML files and folders as component products. No matter whether your YAML file/folder is stored in a Git repository or an OSS bucket, KubeVela can read and deliver it.
```

File: docs/end-user/component-delivery.md [hunk 2]
```diff
@@ -2,4 +2,210 @@
...
+
+Let's take the YAML folder component from the OSS bucket warehouse as an example to explain the usage. In the application deployment plan this time, I hope to deliver a component named bucket-comp. The deployment file corresponding to the component is stored in the cloud storage OSS bucket, and the corresponding bucket name is definition-registry. kustomize.yaml comes from this address of oss-cn-beijing.aliyuncs.com and the path is `./app/prod/`.
```
Comment (FogDong) [inline docs/end-user/component-delivery.md]:
```suggestion
Let's take the YAML folder component from the OSS bucket registry as an example to explain the usage. In the `Application` this time, I hope to deliver a component named bucket-comp. The deployment file corresponding to the component is stored in the cloud storage OSS bucket, and the corresponding bucket name is definition-registry. `kustomize.yaml` comes from this address of oss-cn-beijing.aliyuncs.com and the path is `./app/prod/`.
```

File: docs/end-user/component-delivery.md [hunk 3]
```diff
@@ -2,4 +2,210 @@
...
+
+For usage examples of Git repositories and their detailed configuration item information, please go to [Built-in Components][4] in the Administrator Manual.
+
+## Using Cloud Resource Component
+
+Cloud resource components are also the core components supported by KubeVela. Cloud resources are often not used separately. Cloud resources such as databases and caches are often used after they are created. The relevant information is transferred to other components. At the same time, cloud resources involve different cloud vendors, and Some authentication-related preparations will be explained independently in the chapter [Integrated Cloud Resources][5].
+
+## Custom Component
+
+When none of the above KubeVela's built-in unpacking components can meet your needs, don’t worry, KubeVela provides powerful expansion capabilities and can be connected to almost any type of component form. You can check the [custom components][6] in the administrator manual. ]Learn to use CUE and Kubernetes to extend KubeVela's component types.
```
Comment (hongchaodeng) [inline docs/end-user/component-delivery.md]:
```suggestion
When none of the above KubeVela's built-in unpacking components can meet your needs, don’t worry, KubeVela provides powerful expansion capabilities and can be connected to almost any type of component form. You can check the [custom components][6] in the administrator manual to learn about how to use CUE and Kubernetes to extend KubeVela's component types.
```

File: docs/end-user/component-delivery.md [hunk 4]
```diff
@@ -2,4 +2,210 @@
...
+           	           	                                     	network traffic from customers.                             
+worker     	vela-system	deployments.apps                     	Describes long-running, scalable, containerized services    
+           	           	                                     	that running at backend. They do NOT have network endpoint  
+           	           	                                     	to receive external network traffic.    
+```
+
+Let's take several typical component types as example to introduce the usage of KubeVela component delivery. If you want to directly view how cloud resources are used, please read [Integrate Cloud Service][2].
+
+
+## Using service-based components (webservice)
```
Comment (hongchaodeng) [inline docs/end-user/component-delivery.md]: We should not expose the concept of webservice type to users...

Comment (BinaryHB0916) [reply docs/end-user/component-delivery.md]: you mean we don't have to introduce the using case of webservice, Let them try by themself?

The point of this to tell our users that how they can implement their Micro Service scenarios. Any suggestion on how to change?

File: docs/end-user/component-delivery.md [hunk 5]
```diff
@@ -2,4 +2,210 @@
 title:  Component Delivery
 ---
 
-WIP
\ No newline at end of file
+Components are one of the core concepts of KubeVela. You can use them to build the most common types of services, such as a Web Service that provides external access, a Worker that runs scheduled tasks on the backend, or enables Redis and including OSS from Cloud Provider and so on.
+
+In order to help you quickly and comprehensively implement your business, KubeVela provides a large number of out-of-the-box component types, which not only include business components for conventional microservice scenarios, such as "service-oriented components" (webservice) and "back-end running components" (Worker), “one-off task component” (task), etc., and also include general-purpose components for different product categories in the community, such as the “Helm component” that supports Helm Chart, the “Kustomize component” that supports Git Repo, etc., at last it contains a series of commonly used cloud service components. If KubeVela's built-in component types cannot meet all your needs, KubeVela also has the ability to expand to almost any type of component products.
```
Comment (FogDong) [inline docs/end-user/component-delivery.md]:
```suggestion
In order to help you quickly and comprehensively implement your business, KubeVela provides a large number of out-of-the-box component types, which not only include business components for conventional microservice scenarios, such as "service-oriented components" (webservice) and "back-end running components" (Worker), “
"one-off task component" (task), etc., and also include general-purpose components for different product categories in the community, such as the "Helm component" that supports Helm Chart, the "Kustomize component" that supports Git Repo, etc., at last, it contains a series of commonly used cloud service components. If KubeVela's built-in component types cannot meet all your needs, KubeVela also has the ability to expand to almost any type of component products.
```

File: docs/end-user/component-delivery.md [hunk 6]
```diff
@@ -2,4 +2,210 @@
...
+  services:
+  - healthy: true
+    name: frontend
+    workloadDefinition:
+      apiVersion: apps/v1
+      kind: Deployment
+  status: running
+```
+
+When we see that the status-services-healthy field is true and the status is running, it means that the entire application is delivered successfully.
```
Comment (FogDong) [inline docs/end-user/component-delivery.md]:
```suggestion
When we see that the `status.services.healthy` field is true and the status is running, it means that the entire application is delivered successfully.
```

File: docs/end-user/component-delivery.md [hunk 7]
```diff
@@ -2,4 +2,210 @@
...
+
+If status shows as rendering or healthy as false, it means that the application has either failed to deploy or is still being deployed. Please proceed according to the information returned in `kubectl get application <application name> -o yaml`.
+
+You can also view through the CLI of vela, using the following command:
+```shell
+$ vela ls
+APP    	COMPONENT	TYPE      	TRAITS	PHASE  	HEALTHY	STATUS	CREATED-TIME                 
+website	frontend 	webservice	      	running	healthy	      	2021-08-28 18:26:47 +0800 CST
+```
+We also see that the PHASE of the website APP is running and the STATUS is healthy.
```
Comment (FogDong) [inline docs/end-user/component-delivery.md]:
```suggestion
We also see that the PHASE of the app is running and the STATUS is healthy.
```

File: docs/end-user/component-delivery.md [hunk 8]
```diff
@@ -2,4 +2,210 @@
...
+
+## Using Helm Component
+
+KubeVela's Helm component meets the needs of users to connect to Helm Chart. You can deploy any ready-made Helm chart software package from Helm Repo, Git Repo or OSS bucket through the Helm component, and overwrite its parameters.
+
+We use the Chart package deployment method from the Helm Repo to explain. In this application deployment plan, We hope to deliver a component called redis-comp. It is a Chart from the https://charts.bitnami.com/bitnami Helm repository. The chart type is redis-cluster, version 6.2.7.
```
Comment (FogDong) [inline docs/end-user/component-delivery.md]:
```suggestion
We use the Chart package deployment method from the Helm Repo to explain. In this `Application`, we hope to deliver a component called redis-comp. It is a chart from the [bitnami](https://charts.bitnami.com/bitnami).
```

File: docs/end-user/component-delivery.md [hunk 9]
```diff
@@ -2,4 +2,210 @@
...
+```
+
+Finally, we use `vela ls` to view the application status after successful delivery:
+```shell
+vela ls
+APP                 	COMPONENT  	TYPE      	TRAITS	PHASE  	HEALTHY	STATUS	CREATED-TIME                 
+bucket-app          	bucket-comp	kustomize 	      	running	healthy	      	2021-08-28 18:53:14 +0800 CST
+```
+
+The PHASE of the bucket-app APP is running, and the STATUS is healthy. Successful application deployment!
```
Comment (FogDong) [inline docs/end-user/component-delivery.md]:
```suggestion
The PHASE of the app is running, and the STATUS is healthy. Successful application deployment!
```

**Review by wonderflow — APPROVED** [review]

**Pull request merged**